File: lp2/PokemonV-Java/src/lp2/Player.java
package lp2;

import javax.swing.DefaultListModel;

public class Player extends User {

    private DefaultListModel<Pokemon> pokemons;

    public DefaultListModel<Pokemon> getPokemons() {
        return pokemons;
    }

    public void setPokemons(DefaultListModel<Pokemon> pokemons) {
        this.pokemons = pokemons;
    }

    public Player(String username, String password, String name, String email, int pos_x, int pos_y, Direction direction, DefaultListModel<Item> bag,DefaultListModel<Pokemon> pok){
        super(username,password,name,email,pos_x,pos_y,direction,bag);
        this.pokemons = pok;
    }
    public void catchPokemon() {

    }

    public void catchItem() {

    }
}

File: lp2/PokemonV-C#/PokemonV/Player.cs
using System.ComponentModel;

namespace PokemonV
{
    class Player : User
    {
        public Player(string username, string password, string name, string email) : base(username, password, name, email) {}

        public bool catchPokemon() {
            return true;
        }
        public bool catchItem() {
            return true;
        }

        protected BindingList<Pokemon> pokemons;
        public BindingList<Pokemon> Pokemons
        {
            get { return pokemons; }
            set { pokemons = value; }
        }
    }
}
